feat: support `server.base` configuration (#3542)
web-infra-dev/rsbuild: website/docs/en/guide/basic/server.mdx

Rsbuild comes with a built-in dev server designed to improve the development experience. When you run the `rsbuild dev` or `rsbuild preview` commands, the server will start, providing features such as page preview, routing, and hot module reloading.

## Base Path

By default, the base path of the server is `/`, and users can access output files such as `index.html` and public folder assets through http://localhost:3000/.

> Rsbuild supports modifying the base path of the server through [server.base](/config/server/base). When we want to access these files through http://localhost:3000/foo/, we can configure the following:

Reviewer: ```suggestion
Rsbuild supports modifying the base path of the server through [server.base](/config/server/base). If you to access these files through `http://localhost:3000/foo/`, you can configure the following:
```
